pandas/tests/reshape/test_merge.py

```diff
@@ -1344,32 +1341,194 @@ def f():
+                     Period=['AM', 'PM', 'IP', 'AM', 'OP', 'IP', 'AM'],
+                     LinkType=['a', 'a', 'c', 'b', 'a', 'b', 'a'],
+                     Distance=[100, 80, 90, 80, 75, 35, 55]),
+                columns=['Origin', 'Destination', 'Period',
+                         'LinkType', 'Distance'])
+            .set_index(['Origin', 'Destination', 'Period', 'LinkType']))
+        return right
+
+    # Left join
+    expected_leftj = (
+        DataFrame(
+            dict(Origin=[1, 1, 2, 2, 3],
+                 Destination=[1, 2, 1, 3, 1],
+                 Period=['AM', 'PM', 'IP', 'AM', 'OP'],
+                 TripPurp=['hbw', 'nhb', 'hbo', 'nhb', 'hbw'],
+                 Trips=[1987, 3647, 2470, 4296, 4444],
+                 Distance=[100, 80, 90, np.nan, 75]),
+            columns=['Origin', 'Destination', 'Period',
+                     'TripPurp', 'Trips', 'Distance'])
+        .set_index(['Origin', 'Destination', 'Period', 'TripPurp']))
+
+    # Right join
+    expected_rightj = (
+        DataFrame(
+            dict(Origin=[1, 1, 2, 2, 3, 3, 5],
+                 Destination=[1, 2, 1, 2, 1, 2, 6],
+                 Period=['AM', 'PM', 'IP', 'AM', 'OP', 'IP', 'AM'],
+                 LinkType=['a', 'a', 'c', 'b', 'a', 'b', 'a'],
+                 Trips=[1987, 3647, 2470, np.nan, 4444, np.nan, np.nan],
+                 Distance=[100, 80, 90, 80, 75, 35, 55]),
+            columns=['Origin', 'Destination', 'Period',
+                     'LinkType', 'Trips', 'Distance'])
+        .set_index(['Origin', 'Destination', 'Period', 'LinkType']))
+
+    # Inner join
+    expected_innerj = (
+        DataFrame(
+            dict(Origin=[1, 1, 2, 3],
+                 Destination=[1, 2, 1, 1],
+                 Period=['AM', 'PM', 'IP', 'OP'],
+                 Trips=[1987, 3647, 2470, 4444],
+                 Distance=[100, 80, 90, 75]),
+            columns=['Origin', 'Destination', 'Period',
+                     'Trips', 'Distance'])
+        .set_index(['Origin', 'Destination', 'Period']))
+
+    # Outer join
+    expected_outerj = (
+        DataFrame(
+            dict(Origin=[1, 1, 2, 2, 2, 3, 3, 5],
+                 Destination=[1, 2, 1, 2, 3, 1, 2, 6],
+                 Period=['AM', 'PM', 'IP', 'AM', 'AM', 'OP', 'IP', 'AM'],
+                 TripPurp=['hbw', 'nhb', 'hbo', np.nan, 'nhb',
+                           'hbw', np.nan, np.nan],
+                 LinkType=['a', 'a', 'c', 'b', np.nan, 'a', 'b', 'a'],
+                 Trips=[1987, 3647, 2470, np.nan,
+                        4296, 4444, np.nan, np.nan],
+                 Distance=[100, 80, 90, 80, np.nan, 75, 35, 55]),
+            columns=['Origin', 'Destination', 'Period', 'TripPurp',
+                     'LinkType', 'Trips', 'Distance'])
+        .set_index(['Origin', 'Destination', 'Period',
+                    'TripPurp', 'LinkType']))
+
+    @pytest.mark.parametrize('how, expected', [('left', expected_leftj),
```

Comment:
these you actually DO need to evaluate (e.g. ``expected_left()``) because these are simply calling a function and are not 'parameters' (if they are in the signature they are parameters).